client/my-sites/plans-features-main/index.jsx

```diff
@@ -311,6 +314,12 @@ export class PlansFeaturesMain extends Component {
 			plans = plans.filter( ( planSlug ) => ! isPremiumPlan( planSlug ) );
 		}
 
+		if ( [ 'newsletter', 'link-in-bio' ].includes( flowName ) ) {
```

Comment:
Not a comment for your PR specifically, but we have used this same check in multiple spots. May be time to write hook to consolidate.